src/gentropy/datasource/finngen_ukb_meta/study_index.py

```diff
@@ -0,0 +1,62 @@
+"""Study Index for Finngen data source."""
+from __future__ import annotations
+
+import pyspark.sql.functions as f
+from pyspark.sql import SparkSession
+
+from gentropy.dataset.study_index import StudyIndex
+
+
+class FinngenUkbMetaStudyIndex(StudyIndex):
+    """Study index dataset from FinnGen UKB meta-analysis."""
+
+    @classmethod
+    def from_source(
+        cls: type[FinngenUkbMetaStudyIndex],
+        spark: SparkSession,
+        raw_study_index_path: str,
+    ) -> StudyIndex:
+        """This function ingests study level metadata from FinnGen UKB meta-analysis.
+
+        Args:
+            spark (SparkSession): Spark session object.
+            raw_study_index_path (str): Raw study index path.
+
+        Returns:
+            StudyIndex: Parsed and annotated FinnGen UKB meta-analysis study table.
+        """
+        # Read the raw study index and process.
+        study_index_df = (
+            spark.read.csv(raw_study_index_path, sep="\t", header=True)
+            .select(
+                f.lit("gwas").alias("studyType"),
+                f.lit("FINNGEN_R11_UKB_META").alias("projectId"),
+                f.col("_gentropy_study_id").alias("studyId"),
```

Comment:
Was the source data processed already before this step is run? 
The fields `_gentropy_*` are very specific. Do they come from original datasource?

Reply:
Yes, exactly, the source data for both UKB PPP and this dataset (FinnGen UKB meta-analysis) was processed by code in https://github.com/opentargets/gentropy-input-support

Reply:
Nice!